controller/volume_rebuilding_controller.go

```diff
@@ -96,6 +108,106 @@ func (vbc *VolumeRebuildingController) enqueueVolumeAfter(obj interface{}, durat
 	vbc.queue.AddAfter(key, duration)
 }
 
+func isSettingRelatedToOffLineRebuilding(obj interface{}) bool {
+	setting, ok := obj.(*longhorn.Setting)
+	if !ok {
+		deletedState, ok := obj.(cache.DeletedFinalStateUnknown)
+		if !ok {
+			return false
+		}
+
+		// use the last known state, to enqueue, dependent objects
+		setting, ok = deletedState.Obj.(*longhorn.Setting)
+		if !ok {
+			return false
+		}
+	}
+
+	return setting.Name == string(types.SettingNameOfflineReplicaRebuilding)
+}
+
+func (vbc *VolumeRebuildingController) enqueueSetting(obj interface{}) {
+	vbc.logger.Infof("[James_DBG] enqueueSetting Setting updated: %v in VRBController", obj)
+	_, ok := obj.(*longhorn.Setting)
+	if !ok {
+		deletedState, ok := obj.(cache.DeletedFinalStateUnknown)
+		if !ok {
+			utilruntime.HandleError(fmt.Errorf("received unexpected obj: %#v", obj))
+			return
+		}
+
+		// use the last known state, to requeue the claimed volumes
+		_, ok = deletedState.Obj.(*longhorn.Setting)
+		if !ok {
+			utilruntime.HandleError(fmt.Errorf("DeletedFinalStateUnknown contained non Setting object: %#v", deletedState.Obj))
+			return
+		}
+	}
+
+	offlineRebuildFlag, err := vbc.ds.GetSettingAsBool(types.SettingNameOfflineReplicaRebuilding)
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("failed to get setting %v: %v", types.SettingNameOfflineReplicaRebuilding, err))
+		return
+	}
+
+	vs, err := vbc.ds.ListVolumesRO()
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("failed to list volumes: %v", err))
+		return
+	}
+
+	for _, v := range vs {
+		if v.Spec.OfflineRebuild != longhorn.VolumeOfflineRebuildIgnored {
+			continue
+		}
+		if offlineRebuildFlag && v.Status.State != longhorn.VolumeStateDetached {
+			continue
+		}
+		if !offlineRebuildFlag && v.Status.State != longhorn.VolumeStateAttached {
+			continue
+		}
```

Comment:
Could you clarify the conditions for volume enqueue? Why not just skip:
- `v.Spec.OfflineRebuild != longhorn.VolumeOfflineRebuildIgnored`
- `!offlineRebuildFlag`
- `v.Status.State != longhorn.VolumeStateDetached`

Reply:
Only skip `v.Spec.OfflineRebuild != longhorn.VolumeOfflineRebuildIgnored`.
Others will be handled during the reconciliation.
Thanks